lib/Corres_UL.thy

```diff
@@ -1176,6 +1176,18 @@ lemma corres_move_asm:
 
 lemmas corres_cross_over_guard = corres_move_asm[rotated]
 
+lemma corres_cross_add_guard:
+  "\<lbrakk> \<And>s s'. \<lbrakk>(s,s') \<in> sr; P s; P' s'\<rbrakk> \<Longrightarrow> Q s';
+     corres_underlying sr nf nf' r P  (P' and Q) f g\<rbrakk>
```

Comment:
extra space before `(P'`, also wouldn't `Q'` make more sense than `Q` given it's on the concrete side?

Reply:
Yes it would make much more sense to use `Q'`. This is another rule copied straight over from the `rt` branch. Should I change it here? If I do change it for `master`, should I change it on the `rt` branch too? That would be quite annoying, because there would be many rules which do write `where Q=blah`

Reply:
Do we use `corres_cross_add_guard` that often? Or are you saying there are other rules which also followed the `Q` scheme for the concrete side?

Reply:
`corres_cross_add_guard` is used dozens and dozens of times on the `rt` branch. It wouldn't be horrible to change, but it would be a bit of a nuisance

Reply:
If it's only `corres_cross_add_guard` I'd vote for updating it and its uses, but if other lemmas follow the same scheme, I'd say defer to another round.

Reply:
OK I will update it on both branches

Reply:
Cheers